core/modules/comment/templates/comment.tpl.php

```diff
@@ -64,8 +64,10 @@
   <?php if ($new): ?>
     <mark class="new"><?php print $new; ?></mark>
   <?php endif; ?>
-  <h3><?php print $title; ?></h3>
-  <?php print render($title_suffix); ?>
+  <?php if ($title_options != COMMENT_TITLE_HIDDEN): ?>
```

Comment:
Didn't you want to use an icon instead? This "if" will suppress the title completely if it's "hidden".

Reply:
No, it's a bit more complicated. The core comment.tpl.php file provides it's own permalink, which themes like Basis and several others overwrite. 

PR #1 replaces title with icon with css across the board. Contrib themes can overwrite it. 

PR #2 (this one) simply removes title and let's each theme handle the permalink on it's own, as does Bartik. So in this PR, I simply remove the title in the core and Bartik templates. In Basis, which does not have a separate permalink, I add the icon. This PR (as it currently stands) will not have any effect on contrib themes that overwrite the comment.tpl.php and comment.css files.

Reply:
I'd like to suggest something: can you add (css) classes to that title? Alternative: add a css class to a parent div. That would simplify things for contrib themes.

But probably rather in a preprocessor (add a $title_attributes variable).

Background to this: this feature is based on constants, which will only exist in Backdrop starting with the next minor. If you add things in a preprocessor, themes don't have to also consider if the constants are defined.

Also: AFAIK constants have never been used on core like that (directly in a tpl.php), templates get variables via preprocessors. (TBH, that's how I always do it, I'm not 100% sure if core suggests that.)